### DISCUSSION TITLE
Orleans - failover, scale out

### DISCUSSION THREAD

**Discussion by haragarden3** [dotnet/orleans#8020] [Q&A]
Hi,
I’m completely new to Orleans and my question is perhaps if it is worth diving into in my situation.

I'm in the design phase of creating a “controller” process (service) that posts “work to do” on a message bus and worker processes that can run independently and do the work. The controller needs some simple failover mechanism (one active controller and one passive) and the workers has to be possible to run in 1..n instances. I can solve all of this without Orleans (done that before) but I wonder if I would benefit from introducing Orleans or if it is an overkill. I, for example didn’t find any examples for “simple” failover.

Grateful if I can get some advice 😊

Björn

**Comment by drakeforte5** [comment]
If you use orleans clustering by having the server code run on several hosts  you automatically achieve scale-out and failover functionality.  You might not need a message bus for this as it introduces more complexity.  I was doing a lot of what you were doing using ESB/JMS and my code was certainly more complex due to the handling of state where I had to do this in a batch process like what you are trying to accomplish:

orchestrator ==> workers (1...n instance ) ==> statusupdate

Since orleans has state management in itself my code was simplified.  A different way of thinking on how to solve the problem using orleans for sure.  The keyword for me is "Virtual Actor" which is an embodiment of what an Object is about (state, identity, behavior).

**Reply by drakeforte5** [reply]
A simple sample that can provide confirmation about clustering and failover is this https://learn.microsoft.com/en-us/dotnet/orleans/tutorials-and-samples/tutorial-1

You might have to modify it a bit so you can pass different ports so you can run multiple Silo instances on your dev machine. Then flood it with client requests, kill once silo and observe how the client request is transferred and fulfilled by another silo.
